libs/shared/flagd-core/src/lib/targeting/targeting.ts

```diff
@@ -0,0 +1,31 @@
+import {LogicEngine,} from "json-logic-engine";
+import {endsWithHandler, endsWithRule, startsWithHandler, startsWithRule} from "./string-comp";
+import {semVer, semVerRule} from "./sem-ver";
+import {fractional, fractionalRule} from "./fractional";
+import {flagdPropertyKey, flagKeyPropertyKey, timestampPropertyKey} from "./common";
+
+export class Targeting {
+  private readonly _logicEngine: LogicEngine;
+
+  constructor() {
+    const engine = new LogicEngine();
+    engine.addMethod(startsWithRule, startsWithHandler);
+    engine.addMethod(endsWithRule, endsWithHandler);
+    engine.addMethod(semVerRule, semVer);
+    engine.addMethod(fractionalRule, fractional);
+
+    this._logicEngine = engine;
+  }
+
+  applyTargeting(flagKey: string, logic: any, data: Record<any, any>): any {
+    const ctxData = {
+      [flagdPropertyKey]: {
+        [flagKeyPropertyKey]: flagKey,
+        [timestampPropertyKey]: Date.now(),
```

Comment:
This is ms, not seconds, I think you'd need to change it to be compliant.

Reply:
Thank you. this is a good catch 🙏